compose/project.py

```diff
@@ -619,46 +621,68 @@ def _get_convergence_plans(self, services, strategy, always_recreate_deps=False)
     def pull(self, service_names=None, ignore_pull_failures=False, parallel_pull=False, silent=False,
              include_deps=False):
         services = self.get_services(service_names, include_deps)
-        msg = not silent and 'Pulling' or None
 
         if parallel_pull:
-            def pull_service(service):
-                strm = service.pull(ignore_pull_failures, True, stream=True)
-                if strm is None:  # Attempting to pull service with no `image` key is a no-op
-                    return
+            self.parallel_pull(services, silent=silent)
 
-                writer = parallel.get_stream_writer()
+        else:
+            for service in services:
+                must_build = []
```

Comment:
Shouldn't it be declared outside of the `for`?

Reply:
oups